File: 0139-word-break/python/Solution.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        n = len(s)
        dp = [False] * (n + 1)
        dp[0] = True
        for i in range(1, n+1):
            for word in wordDict:
                lenght = len(word)
                startp = i - lenght
                if startp >= 0 and s.startswith(word,startp) and dp[i - lenght]:
                    dp[i] = True
        return dp[n]

File: 0139-word-break/javascript/Solution.js
/**
 * @param {string} s
 * @param {string[]} wordDict
 * @return {boolean}
 */
var wordBreak = function(s, wordDict) {
    let n = s.length
    let dp = new Array(n+1).fill(false)
    dp[0] = true
    for (let i = 1; i <= n; i++) {
        for (let word of wordDict) {
            let len = word.length
            let startp = i - len
            if (startp >= 0 && s.startsWith(word, startp) && dp[i-len]) {
                dp[i] = true
                break
            }
        }
    }
    return dp[n]
};